turbopack/crates/turbo-tasks/src/manager.rs

```diff
@@ -524,6 +534,9 @@ impl<B: Backend + 'static> TurboTasks<B> {
                 let task_state = RefCell::new(CurrentTaskState::new(
                     task_id,
                     this.execution_id_factory.get(),
+                    this.backend
+                        .try_get_function_id(task_id)
+                        .map(|func_id| &get_function(func_id).function_meta),
```

Comment:
I feel like this should better be returned from `try_start_task_execution` in `TaskExecutionSpec`

Reply:
That does seem like it could be nice in avoiding duplicate lookups of the function, but it looks like `try_start_task_execution` has an implicit dependency on `CURRENT_TASK_STATE`:
- `MemoryBackend::try_start_task_execution` synchronously calls `Task::execute`.
- `Task::execute` synchronously calls `TaskAggregationContext::apply_queued_updates`
- `TaskAggregationContext::apply_queued_updates` synchronously calls `TurboTasks::schedule_notify_tasks_set`
- **`TurboTasks::schedule_notify_tasks_set` modifies `tasks_to_notify` from `CURRENT_TASK_STATE`**

So I can't do this (without significant refactoring) because it would create a circular dependency issue for `CURRENT_TASK_STATE`.